src/ng/rootScope.js

```diff
@@ -504,11 +503,17 @@ function $RootScopeProvider() {
         function watchGroupAction() {
           changeReactionScheduled = false;
 
-          if (firstRun) {
-            firstRun = false;
-            listener(newValues, newValues, self);
-          } else {
-            listener(newValues, oldValues, self);
+          try {
```

Comment:
FWIW, we could avoid the `try/finally` block, by using one extra array. But my intuition is that it won't make any difference.